src/libtiled/mapreader.cpp

```diff
@@ -564,11 +568,48 @@ void MapReaderPrivate::readTilesetTerrainTypes(Tileset &tileset)
     }
 }
 
+void MapReaderPrivate::readTilesetWangSet(Tileset &tileset)
+{
+    Q_ASSERT(xml.isStartElement() && xml.name() == QLatin1String("wangsets"));
+
+    while (xml.readNextStartElement()) {
+        if (xml.name() == QLatin1String("wangset")) {
+            const QXmlStreamAttributes atts = xml.attributes();
+            QString name = atts.value(QLatin1String("name")).toString();
+            int edges = atts.value(QLatin1String("edges")).toInt();
+            int corners = atts.value(QLatin1String("corners")).toInt();
+            int tile = atts.value(QLatin1String("tile")).toInt();
+
+            WangSet *wangSet = new WangSet(&tileset, edges, corners, name, tile);
+
+            tileset.insertWangSet(wangSet);
+
+            while (xml.readNextStartElement()) {
+                if (xml.name() == QLatin1String("properties"))
+                    wangSet->mergeProperties(readProperties());
+                else if (xml.name() == QLatin1String("wangsettile")) {
+                    const QXmlStreamAttributes tileAtts = xml.attributes();
+                    int tileId = tileAtts.value(QLatin1String("tileid")).toInt();
+                    int wangId = tileAtts.value(QLatin1String("wangid")).toInt();
+
+                    wangSet->addTile(tileset.tileAt(tileId), wangId);
```

Comment:
This should be `findOrCreateTile`, since you can't assume the tile is already allocated. In my case this was crashing with your test tileset, since the tiles are usually allocated later, when the image is loaded.